app/quickUnionWeighted.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Created by tatsiak on 4/1/17.
+ */
+'use strict'
+class QuickUnionWeighted {
+    constructor() {
+        this._elements = [];
+        this._size = [];
+    }
+    connect(a, b) {
+        if (this.isConnected(a, b)) return;
+        if (isNaN(this._elements[a])) this._elements[a] = a;
+        if (isNaN(this._elements[b])) this._elements[b] = b;
+
+        let parentA = this.component(a, this._elements);
+        let parentB = this.component(b, this._elements);
+
+        if (isNaN(this._size[parentA])) this._size[parentA] = 1;
+        if (isNaN(this._size[parentB])) this._size[parentB] = 1;
+
+        if (this._size[parentA] > this._size[parentB]) {
```

Comment:
You should connect smaller component to bigger
and you should have a test that checked it